Start testing

zmoon opened the pull request:
* initial dev env to use when testing
* account for the stable `master` branch rename to `stable`
* changes and initial tests for `aeronet`

zmoon committed:
Initial dev env

maybe temporary

zmoon committed:
`import monetio` succeeds!

note that pyresample is currently a required dep in monet

zmoon committed:
aeronet initial tests/updates

zmoon committed:
master branch renamed

zmoon committed:
docstring

zmoon committed:
Add pre-commit GHA

I don't currently have the ability to add pre-commit CI

pep8speaks commented:
Hello @zmoon! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`monetio/obs/aeronet.py`](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py):

> [Line 134:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L134): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 137:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L137): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (92 > 79 characters)
> [Line 148:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L148): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 297:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L297): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 308:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L308): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)
> [Line 321:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L321): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 323:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L323): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)
> [Line 338:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/monetio/obs/aeronet.py#L338): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)



* In the file [`tests/test_aeronet.py`](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/tests/test_aeronet.py):

> [Line 66:80](https://github.com/noaa-oar-arl/monetio/blob/fa48228ae29eda7170bd7861f029cb2f6ab52a53/tests/test_aeronet.py#L66): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)





##### Comment last updated at 2022-01-31 22:21:37 UTC

zmoon committed:
Remove some unused files

zmoon committed:
Initial CI GHA

Squashed commits:

Initial CI GHA

Remove self install from the dev env

another try

give up on mamba for now

zmoon committed:
Validate aeronet sites before requesting

zmoon committed:
Fix aeronet default dates

zmoon committed:
Fix and test getting inv data

zmoon committed:
wip: add_local

zmoon committed:
Fix/test aeronet local file loader

bbakernoaa commented:
@zmoon This is great! I looked through all the file changes. Most are minor and/or changes to make it align with proper python programming 👍 .  I say we merge it
